prov/usnic/src/usdf_pep.c

```diff
@@ -120,7 +122,17 @@ usdf_pep_conn_info(struct usdf_connreq *crp)
 	sin->sin_addr.s_addr = reqp->creq_ipaddr;
 	sin->sin_port = reqp->creq_port;
 
-	ip->dest_addr = sin;
+	if (addr_format_str) {
+		char *addr_str;
+
+		addr_str = calloc(1, USDF_ADDR_STR_LEN);
+		ip->dest_addrlen = sizeof(addr_str);
```

Comment:
This will return sizeof(char *), which is not what is wanted.  You want USDF_ADDR_STR_LEN.